grype/search/distro.go

```diff
@@ -86,3 +86,34 @@ func compatibleVersion(fullVersion string, partialVersion string) bool {
 	}
 	return false
 }
+
+// TODO: this is a temporary workaround... in the long term the mock should more strongly enforce
+// data overrides and not require this kind of logic being baked into mocks directly.
+func mimicV6DistroTypeOverrides(t distro.Type) distro.Type {
```

Comment:
Are there some tests that rely on this? Longer term, I would prefer if we had a separate step the matchers performed to query the provider for OSes, instead of pushing this logic into the stores where it is opaque. I think that could end up being more clear how the overall matching process works and would potentially avoid having this.